pyqtgraph/parametertree/parameterTypes/basetypes.py

```diff
@@ -319,24 +319,37 @@ def __init__(self, param, depth):
         self.optsChanged(self.param, self.param.opts)
 
     def updateDepth(self, depth):
-        ## Change item's appearance based on its depth in the tree
-        ## This allows highest-level groups to be displayed more prominently.
-        if depth == 0:
-            for c in [0,1]:
-                self.setBackground(c, QtGui.QBrush(QtGui.QColor(100,100,100)))
-                self.setForeground(c, QtGui.QBrush(QtGui.QColor(220,220,255)))
-                font = self.font(c)
-                font.setBold(True)
+        """
+        Change item's appearance based on its depth in the tree
+        This allows highest-level groups to be displayed more prominently.
+        """
+        app = QtWidgets.QApplication.instance()
+        palette = app.palette()
+        textColor = palette.text().color()
+        backgroundColor = palette.button().color().darker(110)
+        altBackgroundColor = palette.dark().color()
+
+        # if stylesheet and 'QDarkStyleSheet' in stylesheet:
+        if app.styleSheet(): # Assume the stylesheet is qdarkstyle
+            # #455364 is the background-color of the QHeaderView, #E0E1E3 is lineEdit text from QDarkStyle
+            # lighter/darker are applied to the colors to maintain a good contrast with the item:hover color
+            backgroundColor = QtGui.QColor('#455364').darker(100)
+            altBackgroundColor =  backgroundColor.lighter(130)
+            textColor = QtGui.QColor('#E0E1E3')
+        elif app.property('darkMode'):
+            backgroundColor = palette.button().color().lighter(115)
+            altBackgroundColor = palette.light().color().lighter(190)
```

Comment:
I don't want to complain too much here (Did I say thank you for your work yet? 👍 ), but one awesome thing about a palette is that it makes everything look consistent. If every bit of the application creates its own set of variations on the color set, that is somewhat less effective.
As I already asked above, how far can we get with the base() and alternateBase() colors that seem to exist exactly for this purpose?

Reply:
I responded above, the colors for `base` and `alternatingBase` don't appear as they should. This might only be the case in darkMode on macOS.

Reply:
I looked at the picture you posted, but I couldn't quite figure it out. Could you give me a quick summary of how base and alternateBase colors change between light and dark mode on Mac?
Does base change correctly (dark in dark mode, light in light mode), but alternateBase does not? alternateBase is always that medium gray in your picture?

If that is the case, then that *might* be an argument to move back towards a color adjustment for the less intense alternate color. Not making up a completely new shade, but modifying the palette base shade as little as possible.

Maybe pull `backgroundColor` from the palette and then make the alternative color as
``` python
h, s, l = backgroundColor.getHsl()
l = 0.8 * (l - 0.5) + 0.5 # move towards gray
altBackgroundColor = QColor().setHsl(h,s,l)
```
(for illustration only, there's probably a few errors)
Taking out a little contrast and/or saturation should work for both light and dark schemes, so it should also be able to avoid the case distinction.

Reply:
In Light mode when colors are set to base/alternating base, there are only 2 colors used for backgrounds. In dark mode when I use alternate base, it isn't actually the alternate base color that the tree is using to alternate every other item in the tree, its a third color that is much brighter than the 2 colors the tree is using. 

the alternating base is set on Example Parameters , and base on sample list, sample file, etc. Nothing changes except for the Light/Dark modes
![Screen Shot 67](https://user-images.githubusercontent.com/5187375/142594344-76fdf9e2-ee16-4a77-8b38-e8330fc58eac.png)
![image](https://user-images.githubusercontent.com/5187375/142594513-bcfe8c55-a130-4679-a645-9ad99ee028fa.png)

Reply:
I just went looking for which role Qt is using and it seems to be the Window color in place of alternate base in dark mode.

![Screen Shot 68](https://user-images.githubusercontent.com/5187375/142595262-04c53915-17ed-49a0-ab66-a86a3a966bab.png)